front-end/src/components/Car.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import Tire from "./Tire";
+import Helper from "../utilities/Helper";
+
+class Car extends Component {
+
+    constructor(props) {
+        super(props);
+
+        let data = props.data;
+
+        // Sector 1
+        let sectorOneTime = data.sector_1_time;
+        if (data.sector_1_time === 0)
+            sectorOneTime = data.current_lap_time;
+
+        // Sector 2
+        let sectorTwoTime = data.sector_2_time;
+        if (data.sector_2_time === 0)
+            sectorTwoTime = data.current_lap_time - data.sector_1_time;
+
+        // Sector 3
+        let sectorThreeTime = 0;
+        if (data.sector_1_time && data.sector_2_time)
+            sectorThreeTime = data.current_lap_time - data.sector_1_time - data.sector_2_time;
+
+        // Gap
+        let gap = 0;
+        if (data.car_ahead)
+            gap = data.car_ahead.current_lap_time - data.current_lap_time;
+
+        // Convert time in seconds to a readable format
+        let bestLapTime = Helper.secondsToStr(data.best_lap_time);
+        let currentLapTime = Helper.secondsToStr(data.current_lap_time);
+        sectorOneTime = Helper.secondsToStr(sectorOneTime);
+        sectorTwoTime = Helper.secondsToStr(sectorTwoTime);
+        sectorThreeTime = Helper.secondsToStr(sectorThreeTime);
+        gap = Helper.secondsToStr(gap);
+
+        // Check if in pits
+        let inPits = "";
+        if (data.in_pits === 1) inPits = "Pitting";
+        if (data.in_pits === 2) inPits = "In Pit Area";
+
+        // Initial state
+        this.state = {
```

Comment:
Why is this mixed; eg sometimes a var and sometimes from the data objec, could be always vars or always data object right?

Reply:
These keys could be camelcased as well right?

Reply:
> Why is this mixed; eg sometimes a var and sometimes from the data objec, could be always vars or always data object right?

Sometimes raw data can be shown directly and some data needs to be edited (lap time in seconds to a readable time format)